Neos.Neos/Classes/Domain/Model/WorkspaceRoleAssignments.php

```diff
@@ -33,6 +44,44 @@ public static function fromArray(array $assignments): self
         return new self(...$assignments);
     }
 
+    /**
+     * Default role assignment to be specified at creation via {@see WorkspaceService::createRootWorkspace()}
+     *
+     * Users with the role "Neos.Neos:LivePublisher" are collaborators and everybody can read.
+     */
+    public static function createForLiveWorkspace(): self
+    {
+        return new self(
+            WorkspaceRoleAssignment::createForGroup(
+                'Neos.Neos:LivePublisher',
+                WorkspaceRole::COLLABORATOR
+            ),
+            WorkspaceRoleAssignment::createForGroup(
+                'Neos.Flow:Everybody',
+                WorkspaceRole::VIEWER
+            )
+        );
+    }
```

Comment:
The original idea was to keep those hard-coded role-assignments out of the core leaving it just to the higher levels (Setup, CLI, Workspace module) to define defaults.
Maybe it's more pragmatic to move that to the core though ¯\_(ツ)_/¯

The many magic strings make me a bit nervous, maybe we can introduce an enum or constant for the "special" roles?

Reply:
Actually i tried to improve the current situation as you had `Neos.Neos:LivePublisher` hardcoded _inside_ the service before, now its at least kindof optional if you will, basically a set of sane presets to choose from.
My reasoning was here especially that as the roles are defined inside `Neos.Neos` they ought to be used by this package, otherwise the WorkspaceUi package for example would have the usage to `AbstractEditor` which is odd.

> The many magic strings make me a bit nervous
I like it dangerous :D But yes that how we do it, i mean its originally configuration - unless roles are configured via php we always have that split.

Reply:
As discussed: I'll take care of some follow-up PR that introduces an enum/constants for the "special" roles and replaces the magic strings here and in other places within Neos